src/app/main.rs

```diff
@@ -472,14 +491,20 @@ where
             // 3. Retries are optionally enabled depending on if the route
             //    is retryable.
             let dst_route_layer = svc::builder()
-                .layer(buffer::layer(max_in_flight))
-                .layer(pending::layer())
+                .buffer_pending(max_in_flight, DispatchDeadline::extract)
                 .layer(classify::layer())
                 .layer(metrics::layer::<_, classify::Response>(route_http_metrics))
                 .layer(proxy::http::timeout::layer())
                 .layer(retry::layer(retry_http_metrics.clone()))
                 .layer(metrics::layer::<_, classify::Response>(retry_http_metrics))
-                .layer(insert_target::layer());
+                .layer(insert::target::layer());
+
+            let balancer_stack = svc::builder()
```

Comment:
Eh, this was split out so that it was easier to pin down the errors when it didn't compile. It could be reverted, but also seemed nice and readable this way.

Reply:
I did pretty much the same thing in #240, so I'm definitely +1 on this.